message.go

```diff
@@ -1,86 +1,169 @@
+var defaultCommentTemplate string
+
 type contexts struct {
 	Current  string             `json:"current-context"`
 	Contexts map[string]context `json:"contexts"`
 }
 
 type context struct {
-	Name string `json:"name"`
+	Name *url.URL `json:"name"`
+}
+
+func (p *context) UnmarshalJSON(data []byte) error {
+	type Context context
+
+	tmp := struct {
+		Name string `json:"name"`
+		*Context
+	}{
+		Context: (*Context)(p),
+	}
+
+	err := json.Unmarshal(data, &tmp)
+	if err != nil {
+		return err
+	}
+
+	p.Name, err = url.Parse(tmp.Name)
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
-//Endpoint represents an Okteto statefulset
+// Endpoint represents an Okteto statefulset
 type Endpoint struct {
-	URL     string `json:"url"`
-	Divert  bool   `json:"divert"`
-	Private bool   `json:"private"`
+	URL     *url.URL `json:"url"`
+	Divert  bool     `json:"divert"`
+	Private bool     `json:"private"`
 }
 
-func main() {
-	previewName := os.Args[1]
-	previewCommandExitCode := os.Args[2]
+func (p *Endpoint) UnmarshalJSON(data []byte) error {
+	type endpoint Endpoint
 
-	oktetoURL, err := getOktetoURL()
+	tmp := struct {
+		URL string `json:"url"`
+		*endpoint
+	}{
+		endpoint: (*endpoint)(p),
+	}
+
+	err := json.Unmarshal(data, &tmp)
+	if err != nil {
+		return err
+	}
+
+	p.URL, err = url.Parse(tmp.URL)
 	if err != nil {
-		return
+		return err
 	}
 
-	previewURL := fmt.Sprintf("%s/previews/%s", oktetoURL, previewName)
+	return nil
+}
+
+func generateMessage(previewName string, previewSucceeded bool, commentTemplate string) (string, error) {
+	oktetoURL, err := getOktetoURL()
+	if err != nil {
+		return "", err
+	}
 
-	var firstLine string
-	if previewCommandExitCode == "0" {
-		firstLine = fmt.Sprintf("Your preview environment [%s](%s) has been deployed.", previewName, previewURL)
-	} else {
-		firstLine = fmt.Sprintf("Your preview environment [%s](%s) has been deployed with errors.", previewName, previewURL)
+	commentTemplate, err = getCommentTemplate(commentTemplate)
+	if err != nil {
+		return "", err
 	}
-	fmt.Println(firstLine)
 
 	endpoints, err := getEndpoints(previewName)
 	if err != nil {
-		return
-	}
-	if len(endpoints) == 1 {
-		fmt.Printf("\n  Preview environment endpoint is available [here](%s)", endpoints[0])
-	} else if len(endpoints) > 1 {
-		endpoints = translateEndpoints(endpoints)
-		fmt.Printf("\n  Preview environment endpoints are available at:")
-		for _, endpoint := range endpoints {
-			fmt.Printf("\n  * %s", endpoint)
+		return "", err
+	}
+
+	previewURLSuffix := fmt.Sprintf("%s.%s", previewName, oktetoURL.Host)
+	templateVars := map[string]interface{}{
+		"OktetoURL":        oktetoURL.String(),
+		"PreviewURL":       fmt.Sprintf("%s/#/previews/%s", oktetoURL, previewName),
+		"PreviewName":      previewName,
+		"PreviewURLSuffix": previewURLSuffix,
+		"PreviewSuccess":   previewSucceeded,
+		"Endpoints":        endpoints,
+		"EndpointsMap":     getEndpointsMap(previewURLSuffix, endpoints),
```

Comment:
This is useful in cases like this:
```markdown
### :rocket: [Open MyApp]({{ index .EndpointsMap "app" }}) :rocket:

Supporting services are available at the following links:
- [Minio]({{ index .EndpointsMap "minio" }}) - the minio s3 emulator.
- [Database UI]({{ index .EndpointsMap "adminer" }}) - Database editor.
```
Which would translate to something like this:

> ### :rocket: [Open MyApp](#) :rocket:
> 
> Supporting services are available at the following links:
> - [Minio](#) - the minio s3 emulator.
> - [Database UI](#) - Database editor.